build-tools/tasks/test-utils.js

```diff
@@ -15,24 +16,132 @@ function toWrapper(componentClass) {
 
 const testUtilsSrcDir = path.resolve('src/test-utils');
 const configs = {
+  common: {
+    // These components are not meant to be present in multiple instances in a single app.
+    // For this reason no findAll and findByTestId finders will be generated for them.
+    noExtraFinders: ['AppLayout', 'TopNavigation'],
```

Comment:
A bit more context on this: The "root" element of these two components are buried below the top level element:
```tsx
<div data-testid="assiged-by-user">
  <div>
    <div class="awsui-hashed-root-class">
      ...
    </div>
  </div>
</div>
```

So `data-testid` is not passed to the root element with class name hash, but the top `<div />`. This means `findAppLayoutByTestId` will never return any result and might make the users confused.

Since these two components are also not meant to be multiple in a single app, we can skip adding the extra finders (`findAppLayoutByTestId` and `findAllAppLayouts`) for these ones, at least in the meantime.